.github/workflows/release.yml

```diff
@@ -239,96 +224,64 @@ jobs:
-        run: az acr login -n bitwardenprod
+        run: az acr login -n $_AZ_REGISTRY --only-show-errors
 
       - name: Pull latest project image
-        if: matrix.origin_docker_repo == 'bitwardenprod.azurecr.io'
         env:
           PROJECT_NAME: ${{ steps.setup.outputs.project_name }}
-          ORIGIN_REGISTRY: ${{ matrix.origin_docker_repo }}
         run: |
           if [[ "${{ github.event.inputs.release_type }}" == "Dry Run" ]]; then
-            docker pull $ORIGIN_REGISTRY/$PROJECT_NAME:dev
+            docker pull $_AZ_REGISTRY/$PROJECT_NAME:latest
           else
-            docker pull $ORIGIN_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME
+            docker pull $_AZ_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME
           fi
 
       - name: Tag version and latest
         env:
           PROJECT_NAME: ${{ steps.setup.outputs.project_name }}
-          REGISTRY: bitwardenprod.azurecr.io
-          ORIGIN_REGISTRY: ${{ matrix.origin_docker_repo }}
         run: |
-          if [[ "${{ github.event.inputs.release_type }}" == "Dry Run" ]]; then
-            docker tag $ORIGIN_REGISTRY/$PROJECT_NAME:dev $REGISTRY/$PROJECT_NAME:dryrun
-          else
-            docker tag $ORIGIN_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME $REGISTRY/$PROJECT_NAME:$_RELEASE_VERSION
-            docker tag $ORIGIN_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME $REGISTRY/$PROJECT_NAME:latest
+          if [[ "${{ github.event.inputs.release_type }}" != "Dry Run" ]]; then
+            docker tag $_AZ_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME $_AZ_REGISTRY/$PROJECT_NAME:$_RELEASE_VERSION
+            docker tag $_AZ_REGISTRY/$PROJECT_NAME:$_BRANCH_NAME $_AZ_REGISTRY/$PROJECT_NAME:latest
           fi
 
       - name: Push version and latest image
         if: ${{ github.event.inputs.release_type != 'Dry Run' }}
         env:
           PROJECT_NAME: ${{ steps.setup.outputs.project_name }}
-          REGISTRY: bitwardenprod.azurecr.io
         run: |
-          docker push $REGISTRY/$PROJECT_NAME:$_RELEASE_VERSION
-          docker push $REGISTRY/$PROJECT_NAME:latest
+          docker push $_AZ_REGISTRY/$PROJECT_NAME:$_RELEASE_VERSION
+          docker push $_AZ_REGISTRY/$PROJECT_NAME:latest
 
-      - name: Log out of Docker
-        run: docker logout
+      ########## Push Server to DockerHub ##########
+
+      - name: Setup DCT
+        id: setup-dct
+        if: matrix.project_name == 'Server'
+        uses: bitwarden/gh-actions/setup-docker-trust@74f4ac01c9abe0a7331c9a5de822a558fd4a4710
+        with:
+          azure-creds: ${{ secrets.AZURE_KV_CI_SERVICE_PRINCIPAL }}
+          azure-keyvault-name: "bitwarden-ci"
+
+      - name: Push Server to DockerHub
+        if: ${{ github.event.inputs.release_type != 'Dry Run' && matrix.project_name == 'Server' }}
+        env:
+          PROJECT_NAME: ${{ steps.setup.outputs.project_name }}
+        run: |
+          docker push bitwarden/$PROJECT_NAME:$_RELEASE_VERSION
+          docker push bitwarden/$PROJECT_NAME:latest
```

Comment:
Where are we tagging this to push? Why are we pushing `Server` here instead of in the `self-host` repository with the rest of the images?